src/components/ItemFilters/FiltersContainer.tsx

```diff
@@ -17,62 +23,75 @@ interface ItemFilterContainerProps {
 }
 
 const ItemFilterContainer = ({ itemAggs }: ItemFilterContainerProps) => {
-  const { query } = useRouter()
-  const filterData = itemAggs.map((agg: ItemAggregation) => {
-    if (agg.field === "location") return new LocationFilterData(agg)
-    else return new ItemFilterData(agg)
-  })
-  const [activeFilters, setAppliedFilters] = useState({
-    location: [],
-    format: [],
-    status: [],
-  })
+  const router = useRouter()
+
+  const filterData = useRef(
+    itemAggs.map((agg: ItemAggregation) => {
+      if (agg.field === "location") return new LocationFilterData(agg)
+      else return new ItemFilterData(agg)
+    })
+  ).current
 
+  const appliedFilters = useMemo(() => {
+    return parseItemFilterQueryParams(router.query)
+  }, [router.query])
+
+  const appliedFiltersDisplay = buildAppliedFiltersString(
+    appliedFilters,
+    filterData
+  )
   const ref = useRef<HTMLDivElement>(null)
   useCloseDropDown(() => setWhichFilterIsOpen(""), ref)
 
   const [whichFilterIsOpen, setWhichFilterIsOpen] = useState("")
 
-  const [tempQueryDisplay, setTempQueryDisplay] = useState("")
+  const itemsMatched = buildItemsString(router.query)
```

Comment:
What do you think about renaming the function as `buildItemsMatchedString` or is that too long?